src/Apollo/ReactionsValidator.php

```diff
@@ -0,0 +1,77 @@
+<?php
+declare(strict_types=1);
+
+namespace Triniti\Apollo;
+
+use Gdbots\Pbj\Assertion;
+use Gdbots\Pbj\MessageResolver;
+use Gdbots\Pbj\WellKnown\NodeRef;
+use Gdbots\Pbjx\DependencyInjection\PbjxValidator;
+use Gdbots\Pbjx\Event\PbjxEvent;
+use Gdbots\Pbjx\EventSubscriber;
+use Gdbots\Schemas\Ncr\Enum\NodeStatus;
+
+class ReactionsValidator implements EventSubscriber, PbjxValidator
+{
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            'triniti:apollo:command:add-reactions.validate' => 'validateAddReactions',
+            'triniti:apollo:mixin:reactions.validate'       => 'validateReactions',
+        ];
+    }
+
+    public function validateReactions(PbjxEvent $pbjxEvent): void
+    {
+        $reactions = $pbjxEvent->getMessage();
+
+        if ($reactions->has('reactions') && !empty($reactions->get('reactions'))) {
+            return;
+        }
+
+        foreach($this->getReactions() as $reaction) {
+            $reactions->addToMap('reactions', $reaction, 0);
+        }
+    }
+
+    public function validateAddReactions(PbjxEvent $pbjxEvent): void
+    {
+        $reactions = [];
+
+        $command = $pbjxEvent->getMessage();
+        Assertion::true($this->hasReactions($command->get('node_ref')), 'Node does not support reactions.');
+        Assertion::true($command->has('node_ref'), 'Field "node_ref" is required.');
+        Assertion::true($command->has('reactions'), 'Field "reactions" is required.');
+        Assertion::notEmpty($command->get('reactions'), 'Field "reactions" is empty.');
+
+        $validReactions = $this->getReactions();
+        foreach ($command->get('reactions') as $reaction) {
+            if (in_array($reaction, $validReactions)) {
```

Comment:
@bhsiao55 instead of recreating the reactions, simply throw an exception on the first one that's not valid.

Reply:
@gdbrown do you want to throw an exception on an invalid reaction type or just filter them out?  I was thinking we just filter the invalid ones from the command.